packages/contracts/src/diamond/facets/ManagerFacet.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import {
+    AppStorage,
+    Modifiers,
+    UBQ_MINTER_ROLE,
+    PAUSER_ROLE,
+    COUPON_MANAGER_ROLE,
+    BONDING_MANAGER_ROLE,
+    INCENTIVE_MANAGER_ROLE,
+    UBQ_TOKEN_MANAGER_ROLE    
+} from "../libraries/LibAppStorage.sol";
+import { AccessControlStorage } from "solidstate/contracts/access/access_control/AccessControlStorage.sol";
```

Comment:
I am sure we can do this without solidstate library. but if you found anything which couldn't do that without them, please make us clear why we SHOULD need.

Reply:
> I am sure we can do this without solidstate library. but if you found anything which couldn't do that without them, please make us clear why we SHOULD need.

Yeah, totally agree with you. I can simply abstract the access control part of the solidstate library for our access control system and remove it. But it is fully designed for upgradable smart contract, so we might get much useful codebase from it
And I'm saying again that Openzepplin library can't be used for diamond implementation

Reply:
Why do we need another upgradable smart contract, Diamond is a upgradable smart contract pattern itself.